src/components/user/services/userProfile.service.js

```diff
@@ -70,19 +70,34 @@ angular.module('znk.infra.user').service('UserProfileService',
         }
 
         function _setProfile(newProfile, userId) {
+            var saveProfileProm = [];
             var authData = AuthService.getAuth();
             if (authData || userId){
                 var uid = userId ? userId : authData.uid;
                 var profilePath = 'users/' + uid + '/profile';
                 return UserStorageService.get(profilePath).then(function (profile) {
-                    return profile ? UserStorageService.update(profilePath, newProfile) : UserStorageService.set(profilePath, newProfile);
+                    if (ENV.setUserProfileTwice) {
+                        saveProfileProm.push(_setUserProfileTwice(profilePath, newProfile));
+                    }
+                    if (profile){
+                        saveProfileProm.push(UserStorageService.update(profilePath, newProfile));
+                    } else {
+                        saveProfileProm.push(UserStorageService.set(profilePath, newProfile));
+                    }
+                    return $q.all(saveProfileProm);
                 });
             } else {
                 $log.error('UserProfileService.setProfile: No user were found');
                 return $q.when(null);
             }
         }
 
+        function _setUserProfileTwice(profilePath, newProfile) {
+            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
+                return globalStorage.set(profilePath, newProfile);
```

Comment:
I know that's way it's called _setUserProfileTwice.
once to UserProfileService (znk-dev) twice to globalStorage